controllers/gateway/apirule_controller.go

```diff
@@ -118,6 +118,20 @@ func (r *APIRuleReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 		DefaultDomainName: defaultDomainName,
 	}
 
+	convertedApiRule := &gatewayv1beta1.APIRule{}
+	err = r.Client.Get(ctx, req.NamespacedName, convertedApiRule)
+	// If the ApiRule is not found, we don't need to do anything. If it's found and converted, CM reconciliation is not needed.
+	if err == nil && convertedApiRule.Annotations != nil {
+		if originalVersion, ok := convertedApiRule.Annotations["gateway.kyma-project.io/original-version"]; ok && originalVersion == "v1beta2" {
+			r.Log.Info("ApiRule is converted from v1beta2")
+			if r.Config.JWTHandler == helpers.JWT_HANDLER_ORY {
+				r.Log.Info("APIRule in version v1beta2 is not supported with Ory handler.")
+				return doneReconcileNoRequeue()
+			}
+			r.Config.JWTHandler = helpers.JWT_HANDLER_ISTIO
```

Comment:
Why are setting this as the default here? Wouldn't this conflict with [this code](https://github.com/werdes72/api-gateway/blob/e8a1ac188fbe21938e8fb0bb73298c8696f8bdaa/controllers/gateway/apirule_controller.go#L136)?

I never liked that we mixed up APIRule and ConfigMap reconciliation in one controller and this makes it now a little bit worse. I think we should move this code to the actual "start of APIRule reconciliation" [here](https://github.com/werdes72/api-gateway/blob/e8a1ac188fbe21938e8fb0bb73298c8696f8bdaa/controllers/gateway/apirule_controller.go#L159).

Did you think about extracting the new code to a function, so we don't extend the length of this function, since there were already complains about it?

Reply:
It will disable the CM reconciliation when APIRule is in v1beta2 version.
Moved to separate function.